Sources/Yams/Representer.swift

```diff
@@ -192,38 +200,71 @@ extension Optional: NodeRepresentable {
 }
 
 extension Decimal: ScalarRepresentable {
-    public func represented() throws -> Node {
-        return Node(description)
+    public func represented() -> Node.Scalar {
+        return .init(description)
     }
 }
 
 extension URL: ScalarRepresentable {
-    public func represented() throws -> Node {
-        return Node(absoluteString)
+    public func represented() -> Node.Scalar {
+        return .init(absoluteString)
+    }
+}
+
+extension String: ScalarRepresentable {
+    public func represented() -> Node.Scalar {
+        return .init(self)
     }
 }
 
 /// MARK: - ScalarRepresentableCustomizedForCodable
 
-public protocol ScalarRepresentableCustomizedForCodable: ScalarRepresentable {
-    func representedForCodable() -> Node
+public protocol YAMLEncodable: Encodable {
+    func box() -> Node
 }
 
-extension Date: ScalarRepresentableCustomizedForCodable {
-    public func representedForCodable() -> Node {
+extension YAMLEncodable where Self: ScalarRepresentable {
+    public func box() -> Node {
+        return .scalar(represented())
+    }
+}
+
+extension Bool: YAMLEncodable {}
+extension Data: YAMLEncodable {}
+extension Decimal: YAMLEncodable {}
+extension Int: YAMLEncodable {}
```

Comment:
Can we do `extension BinaryInteger: YAMLEncodable` or `extension Numeric: YAMLEncodable`?

Reply:
Swift compiler does not allow writing extension of protocol to adding conformance.
```
Yams/Sources/Yams/Representer.swift:235:1: error: extension of protocol 'BinaryInteger' cannot have an inheritance clause
extension BinaryInteger: YAMLEncodable {
^                        ~~~~~~~~~~~~~
```

Reply:
Ah, right I forgot.